pkg/frontend/transport/handler.go

```diff
@@ -41,6 +41,35 @@ var (
 	errRequestEntityTooLarge = httpgrpc.Errorf(http.StatusRequestEntityTooLarge, "http: request body too large")
 )
 
+const (
+	reasonRequestBodyTooLarge     = "request_body_too_large"
+	reasonResponseTooLarge        = "response_too_large"
+	reasonTooManyRequests         = "too_many_requests"
+	reasonTooLongRange            = "too_long_range"
+	reasonTooManySamples          = "too_many_samples"
+	reasonSeriesFetched           = "series_fetched"
```

Comment:
Nit: I think it would be more consistent if we use `too_many_series_fetched`. I am thinking about `cortex_discarded_queries_total{reason="series_fetched"}` v.s. `cortex_discarded_queries_total{reason="too_many_series_fetched"}`; I think later one is more clear to reader.

However, the counter argument is that the label name become bigger and may have performance impact. In which case maybe we should change `too_many_samples` to `samples`?

Reply:
`too_many_` is kind of redundant I feel as we usually don't limit `too_few`. I will try to rename a few but I have to keep `too_many_requests` and `too_many_samples`. It is not easy to find a better name